### PR TITLE
Various barclamp_install fixes

### PR DESCRIPTION
**Pull request by rsalevsky**

### CODE REVIEW

**Comment by tboerger** [comment]
:+1:

File: scripts/barclamp_install.rb [hunk 1]
```diff
@@ -124,7 +124,7 @@ def usage()
       target=File.join(BARCLAMP_PATH, bc[:src].split("/")[-1])
       if File.directory? target
         debug "target directory #{target} exists"
-        if get_crowbar_yml_path(target).nil?
+        if get_yml_paths(target, bc[:name]).nil?
```
Comment (vuntz) [inline scripts/barclamp_install.rb]: This will never be nil, but maybe an empty array...

File: scripts/barclamp_mgmt_lib.rb [hunk 2]
```diff
@@ -440,7 +440,7 @@ def bc_install_layout_1_app(from_rpm, bc, bc_path, yaml)
   yml_path = File.join CROWBAR_PATH, 'barclamps'
   yml_created = File.join(yml_path, "#{bc}.yml")
   FileUtils.mkdir yml_path unless File.directory? yml_path
-  FileUtils.cp yml_barclamp, yml_created
+  FileUtils.cp yml_barclamp, yml_created unless yml_barclamp == yml_created
```
Comment (vuntz) [inline scripts/barclamp_mgmt_lib.rb]: Hrm, when would this happen?

Comment (rsalevsky) [inline scripts/barclamp_mgmt_lib.rb]: It happens when this https://github.com/crowbar/crowbar/blob/master/scripts/barclamp_install.rb#L165-L170 is triggert.

Comment (vuntz) [inline scripts/barclamp_mgmt_lib.rb]: Thanks

**Comment by vuntz** [comment]
+1

File: scripts/barclamp_install.rb [hunk 3]
```diff
@@ -124,7 +124,7 @@ def usage()
       target=File.join(BARCLAMP_PATH, bc[:src].split("/")[-1])
       if File.directory? target
         debug "target directory #{target} exists"
-        if get_crowbar_yml_path(target).nil?
+        if get_yml_paths(target, bc[:name]).empty?
           debug "crowbar YAML file does not exists in #{target}"
```
Comment (tboerger) [inline scripts/barclamp_install.rb]: Than you need to update this sentence as well.

Comment (vuntz) [inline scripts/barclamp_install.rb]: Why?

Comment (tboerger) [inline scripts/barclamp_install.rb]: Forget it

**Comment by dguitarbite** [comment]
+1

**Pull request merged**